Src/Particle/AMReX_ParticleContainer.H

```diff
@@ -1244,6 +1244,7 @@ public:
         m_num_runtime_real++;
         h_redistribute_real_comp.push_back(communicate);
         SetParticleSize();
+        this->resizeData();
```

Comment:
@atmyers potentially, you could also move the logic in `AddRealComp` that reads `//resize runtime SoA` into `resizeData`.